engine/src/main/java/org/hibernate/validator/messageinterpolation/ExpressionLanguageFeatureLevel.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.messageinterpolation;
+
+import org.hibernate.validator.Incubating;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+
+/**
+ * Indicates the level of features enabled for the Expression Language engine.
+ *
+ * @since 6.2
+ */
+@Incubating
+public enum ExpressionLanguageFeatureLevel {
+
+	/**
+	 * Expression Language expressions are not interpolated.
+	 */
+	NONE,
+
+	/**
+	 * Only allows access to the variables injected via
+	 * {@link HibernateConstraintValidatorContext#addExpressionVariable(String, Object)}, the Jakarta Bean
+	 * Validation-defined {@code formatter} and the {@code ResourceBundle}s.
+	 */
+	SAFEST,
+
+	/**
+	 * Only allows to access bean properties and the Jakarta Bean Validation-defined {@code formatter}.
```

Comment:
No access to the expression variables and resource bundle? In either case, I think this should be stated explicitly.